### ISSUE TITLE
Loss exploding despite using Pretrained model.

### ISSUE THREAD

**Issue by akshitamittel** [sfzhang15/RefineDet#73]
I was using the trainval35k (VGG-16): [RefineDet320](https://drive.google.com/file/d/1BIVvkSQM8DarpeqOpjgPpDxUoFq9ZxKM/view) model to initialize the network given in examples/refindet/VGG16_COCO_320.py as a simple sanity check for my experiment.

I have used all three learning rates, (1e-3, 1e-4, 1e-5) along with the multi-step learning rate in various experiments, but my loss keeps exploding right after the first few iterations. A sample of my initial log is as follows:

```
3356 I0606 15:51:53.162600 25269 solver.cpp:243] Iteration 0, loss = 3.04811
3357 I0606 15:51:53.162643 25269 solver.cpp:259]     Train net output #0: arm_loss = 1.73809 (* 1 = 1.73809 loss)
3358 I0606 15:51:53.162652 25269 solver.cpp:259]     Train net output #1: odm_loss = 1.31003 (* 1 = 1.31003 loss)
3359 I0606 15:51:53.593672 25269 sgd_solver.cpp:138] Iteration 0, lr = 1e-05
3360 I0606 15:52:24.685073 25269 solver.cpp:243] Iteration 10, loss = 6.96424
3361 I0606 15:52:24.685175 25269 solver.cpp:259]     Train net output #0: arm_loss = 2.54221 (* 1 = 2.54221 loss)
3362 I0606 15:52:24.685186 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.09681 (* 1 = 2.09681 loss)
3363 I0606 15:52:26.588045 25269 sgd_solver.cpp:138] Iteration 10, lr = 1e-05
3364 I0606 15:52:52.352267 25269 solver.cpp:243] Iteration 20, loss = 6.33065
3365 I0606 15:52:52.352313 25269 solver.cpp:259]     Train net output #0: arm_loss = 6.78254 (* 1 = 6.78254 loss)
3366 I0606 15:52:52.352322 25269 solver.cpp:259]     Train net output #1: odm_loss = 3.04511 (* 1 = 3.04511 loss)
3367 I0606 15:52:53.479641 25269 sgd_solver.cpp:138] Iteration 20, lr = 1e-05
3368 I0606 15:53:17.466629 25269 solver.cpp:243] Iteration 30, loss = 5.1626
3369 I0606 15:53:17.466894 25269 solver.cpp:259]     Train net output #0: arm_loss = 2.92764 (* 1 = 2.92764 loss)
3370 I0606 15:53:17.466910 25269 solver.cpp:259]     Train net output #1: odm_loss = 3.68891 (* 1 = 3.68891 loss)
3371 I0606 15:53:18.822995 25269 sgd_solver.cpp:138] Iteration 30, lr = 1e-05
3372 I0606 15:53:42.499008 25269 solver.cpp:243] Iteration 40, loss = 6.90341
3373 I0606 15:53:42.499053 25269 solver.cpp:259]     Train net output #0: arm_loss = 3.95934 (* 1 = 3.95934 loss)
3374 I0606 15:53:42.499061 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.74733 (* 1 = 2.74733 loss)
3375 I0606 15:53:44.003513 25269 sgd_solver.cpp:138] Iteration 40, lr = 1e-05
3376 I0606 15:54:05.387426 25269 solver.cpp:243] Iteration 50, loss = 6.57755
3377 I0606 15:54:05.387586 25269 solver.cpp:259]     Train net output #0: arm_loss = 3.73033 (* 1 = 3.73033 loss)
3378 I0606 15:54:05.387598 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.9674 (* 1 = 2.9674 loss)
3379 I0606 15:54:07.009681 25269 sgd_solver.cpp:138] Iteration 50, lr = 1e-05
3380 I0606 15:54:33.022603 25269 solver.cpp:243] Iteration 60, loss = 6.68664
3381 I0606 15:54:33.022656 25269 solver.cpp:259]     Train net output #0: arm_loss = 7.10667 (* 1 = 7.10667 loss)
3382 I0606 15:54:33.022665 25269 solver.cpp:259]     Train net output #1: odm_loss = 4.41415 (* 1 = 4.41415 loss)
3383 I0606 15:54:33.967619 25269 sgd_solver.cpp:138] Iteration 60, lr = 1e-05
3384 I0606 15:54:58.932065 25269 solver.cpp:243] Iteration 70, loss = 6.87257
3385 I0606 15:54:58.932274 25269 solver.cpp:259]     Train net output #0: arm_loss = 2.7369 (* 1 = 2.7369 loss)
3386 I0606 15:54:58.932286 25269 solver.cpp:259]     Train net output #1: odm_loss = 1.93219 (* 1 = 1.93219 loss)
3387 I0606 15:54:59.851728 25269 sgd_solver.cpp:138] Iteration 70, lr = 1e-05
3388 I0606 15:55:22.794116 25269 solver.cpp:243] Iteration 80, loss = 6.38935
3389 I0606 15:55:22.794162 25269 solver.cpp:259]     Train net output #0: arm_loss = 2.39408 (* 1 = 2.39408 loss)
3390 I0606 15:55:22.794170 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.48458 (* 1 = 2.48458 loss)
3391 I0606 15:55:24.332151 25269 sgd_solver.cpp:138] Iteration 80, lr = 1e-05
3392 I0606 15:55:49.092753 25269 solver.cpp:243] Iteration 90, loss = 7.31767
3393 I0606 15:55:49.092905 25269 solver.cpp:259]     Train net output #0: arm_loss = 3.08944 (* 1 = 3.08944 loss)
3394 I0606 15:55:49.092916 25269 solver.cpp:259]     Train net output #1: odm_loss = 3.70868 (* 1 = 3.70868 loss)
3395 I0606 15:55:51.067596 25269 sgd_solver.cpp:138] Iteration 90, lr = 1e-05
3396 I0606 15:56:16.370951 25269 solver.cpp:243] Iteration 100, loss = 7.69724
3397 I0606 15:56:16.371011 25269 solver.cpp:259]     Train net output #0: arm_loss = 5.55732 (* 1 = 5.55732 loss)
3398 I0606 15:56:16.371019 25269 solver.cpp:259]     Train net output #1: odm_loss = 4.7402 (* 1 = 4.7402 loss)
3399 I0606 15:56:18.080432 25269 sgd_solver.cpp:138] Iteration 100, lr = 1e-05
3400 I0606 15:56:45.454825 25269 solver.cpp:243] Iteration 110, loss = 6.52994
3401 I0606 15:56:45.455013 25269 solver.cpp:259]     Train net output #0: arm_loss = 5.13252 (* 1 = 5.13252 loss)
3402 I0606 15:56:45.455024 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.26274 (* 1 = 2.26274 loss)
3403 I0606 15:56:46.395331 25269 sgd_solver.cpp:138] Iteration 110, lr = 1e-05
3404 I0606 15:57:08.362797 25269 solver.cpp:243] Iteration 120, loss = 6.79823
3405 I0606 15:57:08.362849 25269 solver.cpp:259]     Train net output #0: arm_loss = 3.82763 (* 1 = 3.82763 loss)
3406 I0606 15:57:08.362859 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.85614 (* 1 = 2.85614 loss)
3407 I0606 15:57:08.842171 25269 sgd_solver.cpp:138] Iteration 120, lr = 1e-05
3408 I0606 15:57:34.687839 25269 solver.cpp:243] Iteration 130, loss = 6.21087
3409 I0606 15:57:34.687983 25269 solver.cpp:259]     Train net output #0: arm_loss = 2.7597 (* 1 = 2.7597 loss)
3410 I0606 15:57:34.687994 25269 solver.cpp:259]     Train net output #1: odm_loss = 1.34279 (* 1 = 1.34279 loss)
3411 I0606 15:57:35.269722 25269 sgd_solver.cpp:138] Iteration 130, lr = 1e-05
3412 I0606 15:57:57.090904 25269 solver.cpp:243] Iteration 140, loss = 6.71005
3413 I0606 15:57:57.090953 25269 solver.cpp:259]     Train net output #0: arm_loss = 4.80431 (* 1 = 4.80431 loss)
3414 I0606 15:57:57.090961 25269 solver.cpp:259]     Train net output #1: odm_loss = 3.42285 (* 1 = 3.42285 loss)
3415 I0606 15:57:58.072175 25269 sgd_solver.cpp:138] Iteration 140, lr = 1e-05
3416 I0606 15:58:21.745090 25269 solver.cpp:243] Iteration 150, loss = 6.84413
3417 I0606 15:58:21.745301 25269 solver.cpp:259]     Train net output #0: arm_loss = 3.6142 (* 1 = 3.6142 loss)
3418 I0606 15:58:21.745311 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.40951 (* 1 = 2.40951 loss)
3419 I0606 15:58:22.705026 25269 sgd_solver.cpp:138] Iteration 150, lr = 1e-05
3420 I0606 15:58:46.629601 25269 solver.cpp:243] Iteration 160, loss = 7.42296
3421 I0606 15:58:46.629657 25269 solver.cpp:259]     Train net output #0: arm_loss = 3.39312 (* 1 = 3.39312 loss)
3422 I0606 15:58:46.629669 25269 solver.cpp:259]     Train net output #1: odm_loss = 2.92923 (* 1 = 2.92923 loss)
3423 I0606 15:58:48.365651 25269 sgd_solver.cpp:138] Iteration 160, lr = 1e-05
3424 I0606 15:59:09.365672 25269 solver.cpp:243] Iteration 170, loss = 6.78388
3425 I0606 15:59:09.365896 25269 solver.cpp:259]     Train net output #0: arm_loss = 5.21783 (* 1 = 5.21783 loss)
3426 I0606 15:59:09.365908 25269 solver.cpp:259]     Train net output #1: odm_loss = 3.34821 (* 1 = 3.34821 loss)
3427 I0606 15:59:11.833418 25269 sgd_solver.cpp:138] Iteration 170, lr = 1e-05
3428 I0606 15:59:38.078543 25269 solver.cpp:243] Iteration 180, loss = 6.34393
3429 I0606 15:59:38.078584 25269 solver.cpp:259]     Train net output #0: arm_loss = 2.64921 (* 1 = 2.64921 loss)
3430 I0606 15:59:38.078595 25269 solver.cpp:259]     Train net output #1: odm_loss = 3.13618 (* 1 = 3.13618 loss)
3431 I0606 15:59:39.236337 25269 sgd_solver.cpp:138] Iteration 180, lr = 1e-05
3432 I0606 16:00:04.054961 25269 solver.cpp:243] Iteration 190, loss = 6.44041
```

Is this normal, or is there an issue with say the lmdb file that I have created. (I am following the instructions that were given to generate the COCO dataset). 

Thank you for your help!

**Comment by sfzhang15** [comment]
@akshitamittel Hi,
I think it is not normal, since the pretrained model (VGG16_320_COCO) you use to initialize the network already has low loss, and your loss keeps exploding right after the first few iterations even with very low learning rate (1e-05). You should firstly check your LMDB and I can sent my lmdb to you, but it is very large. Except the lmdb, anything else has been changed?

**Comment by akshitamittel** [comment]
No, for this particular experiment I used your script as it is (apart from loading the pretrained model). I also ran a similar set of experiments with VOCPlus in another example, which seems to be fine (the loss is between 2-3.5). 

I did follow the lmdb file creation for COCO to the dot. I understand that; my train_lmdb file for COCO is above 19G too.

**Comment by akshitamittel** [comment]
Are there any sanity checks for an lmdb file?

**Comment by sfzhang15** [comment]
@akshitamittel 
My train_lmdb file for COCO is 19.6 GB (19,596,865,536 bytes). As for the sanity checks, I usually use a python file to read the images and bboxes from the lmdb, then display them to see whether it is right. Besides, you can provide your training log file and I can check it for you.

**Comment by akshitamittel** [comment]
Sure, I will send you my log files shortly. Thank you so much for this!
